### PR TITLE
PyGame or PyAudio support

### PR DESCRIPTION
**Pull request by kkirchhoff01**
Added PyAudio support if PyGame isn't available so people running OS X with PyAudio can get audio without having to use a different file. I haven't tried this on a Mac, so if someone has a Mac with the PyGame problem please test this and merge it or fix it if it doesn't work.

### CODE REVIEW

File: pyre.py [hunk 1]
```diff
@@ -111,6 +151,14 @@ def redraw(self):
   optlist, args = getopt.getopt(sys.argv[1:], 's:r:i:w:h:')
   fire = Fire(dict(optlist))
 
+  def signal_handler(signal, frame):
```
Comment (emgram769) [inline pyre.py]: This shouldn't be necessary anymore, right?

**Comment by kkirchhoff01** [comment]
OK, I removed that. There were a few errors with pyaudio in my original pull request. It should work now, cross platform with python 2 and 3.

File: pyre.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/python
 import random
+import signal
```
Comment (emgram769) [inline pyre.py]: no longer necessary

File: pyre.py [hunk 3]
```diff
@@ -47,9 +51,46 @@ def color(r, g, b):
     assert(len(self.particles) == self.NUM_PARTICLES)
 
     self.resize()
+    self.volume = 1.0
+    if pygame_available:
+      mixer.init()
+      music.load('fire.wav')
+      music.play(-1)
+    elif pygame_available == False and pyaudio_available:
```
Comment (emgram769) [inline pyre.py]: `elif` implies `pygame_available == False`, perhaps make this line
`elif pyaudio_available:` for conciseness

File: pyre.py [hunk 4]
```diff
@@ -47,9 +51,46 @@ def color(r, g, b):
     assert(len(self.particles) == self.NUM_PARTICLES)
 
     self.resize()
+    self.volume = 1.0
+    if pygame_available:
```
Comment (emgram769) [inline pyre.py:54]: not necessary for this commit, but do you think we should move the audio checking logic into `play_fire`? might be cleaner. thoughts?

Comment (kkirchhoff01) [inline pyre.py:54]: That would require threading to be imported when pyaudio isn't being used, which seems unnecessary.

Comment (emgram769) [inline pyre.py:54]: well, the threading logic would have to be moved to that function as well

**Pull request merged**